### ISSUE TITLE
Relative callback URL

### ISSUE THREAD

**Issue by TheBlueMatt** [btcontract/lnurl-rfc#134]
The callback url in payRequest makes it impossible to create a static implementation of a payRequest server - the server must know its full URI to return in the callback parameter. It would be nice to be able to specify at least server-relative callback URLs so that a static implementation (run behind NGINX or some other TLS proxy) could be built that doesn't need additional LNURL-specific configuration

**Comment by fiatjaf** [comment]
I agree.

**Comment by fiatjaf** [comment]
Even though I agree and think it was a design mistake to require full URLs, in most cases you can get away with that on the server side by reading `Host`, `X-Forwarded-Host` and all the other million special headers HTTP tools have for these purposes and then rebuilding the original URL of the first request.

**Comment by hsjoberg** [comment]
I agree, this would be great for easing Lightning Address deployment.

It sounds like it would be backwards incompatible for most if not all wallet implementations.

**Comment by andrerfneves** [comment]
Closing stale (3+ yrs) issue. Agree this is a limitation on the spec but as pointed above it would be backwards incompatible for most if not all wallets.
